module/zfs/vdev_raidz.c

```diff
@@ -707,8 +709,21 @@ vdev_raidz_generate_parity_pq(raidz_map_t *rm)
 		ccnt = rm->rm_col[c].rc_size / sizeof (p[0]);
 
 		if (c == rm->rm_firstdatacol) {
-			abd_copy_to_buf(p, src, rm->rm_col[c].rc_size);
-			(void) memcpy(q, p, rm->rm_col[c].rc_size);
+			/*
+			 * Guard against zero size, which in DEBUG will
+                         * cause an ASSERTion in abd_copy_to_buf
+			 */
+			if (rm->rm_col[c].rc_size > 0) {
+				ASSERT3S(src->abd_size,>=,rm->rm_col[c].rc_size);
+				abd_copy_to_buf_off(p, src, 0, rm->rm_col[c].rc_size);
+				(void) memcpy(q, p, rm->rm_col[c].rc_size);
+#ifndef DEBUG
+			}
+#else
+		        } else {
+			        printf("%s: rm->rm_col[c].rc_size == 0, skipping\n", __func__);
```

Comment:
This needs to be revisited.   It can probably be demoted to a dprintf, but it's a live issue in "minefield" pool (also see comment above):

sudo ./cmd.sh zdb -c -L -U /etc/zfs/caches/Dual.cache -V Dual
vdev_file_io_start: (pseudo) Assertion: zio_size_in 1000 != abd_size_in 2000
vdev_file_io_start: (pseudo) Assertion: zio_size_in 1000 != abd_size_in 3000
vdev_file_io_start: (pseudo) Assertion: zio_size_in 1000 != abd_size_in 4000
vdev_file_io_start: (pseudo) Assertion: zio_size_in 1000 != abd_size_in 2000
vdev_file_io_start: (pseudo) Assertion: zio_size_in 1000 != abd_size_in 3000
vdev_file_io_start: (pseudo) Assertion: zio_size_in 1000 != abd_size_in 4000
vdev_raidz_generate_parity_pq: rm->rm_col[c].rc_size == 0, skipping
vdev_raidz_generate_parity_pq: rm->rm_col[c].rc_size == 0, skipping
vdev_raidz_generate_parity_pq: rm->rm_col[c].rc_size == 0, skipping
vdev_raidz_generate_parity_pq: rm->rm_col[c].rc_size == 0, skipping
vdev_raidz_generate_parity_pq: rm->rm_col[c].rc_size == 0, skipping
vdev_raidz_generate_parity_pq: rm->rm_col[c].rc_size == 0, skipping